src/client/scripts/esm/chess/variants/variant.ts

```diff
@@ -59,19 +59,19 @@ interface GameRuleModifications {
 	promotionRanks?: (number | null)[] | null,
 	moveRule?: number | null,
 	turnOrder?: string[],
-	promotionsAllowed?: {
-		[key: string]: string[]
-	},
-	winConditions?: {
-		[key: string]: string[]
-	},
+	promotionsAllowed?: ColorVariantProperty<string[]>
+	winConditions?: ColorVariantProperty<string[]>
 	slideLimit?: number
 }
 
 type TimeVariantProperty<T> = T | {
 	[key: number]: T
 }
 
+type ColorVariantProperty<T> = {
```

Comment:
Perhaps add comments to this and TimeVariantProperty of what their keys should be. For example, color, and a timestamp for the other, or an object of timestamps for the keys